pkg/api/meta/namespace.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meta
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// GetNamespace returns the namespace for the given object, if there is one. If not, returns
+// the empty string and a non-nil error
+func GetNamespace(obj runtime.Object) (string, error) {
+	return selfLinker.Namespace(obj)
```

Comment:
no good reason except that more code used the `selfLinker` before this patch, so I've left this as-is